src/main/java/com/synopsys/integration/detect/workflow/diagnostic/DiagnosticsDecider.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * synopsys-detect
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.detect.workflow.diagnostic;
+
+import com.synopsys.integration.configuration.config.PropertyConfiguration;
+import com.synopsys.integration.detect.configuration.DetectProperties;
+import com.synopsys.integration.detect.help.DetectArgumentState;
+
+public class DiagnosticsDecider {
+    private final DetectArgumentState detectArgumentState;
+    private final PropertyConfiguration propertyConfiguration;
+
+    public DiagnosticsDecider(DetectArgumentState detectArgumentState, PropertyConfiguration propertyConfiguration) {
+        this.detectArgumentState = detectArgumentState;
+        this.propertyConfiguration = propertyConfiguration;
+    }
+
+    public boolean isDiagnostic() {
```

Comment:
Did you consider having a single method that returned a result object?

Reply:
I've made `DiagnosticsDecider` return a `DiagnosticsDecision`.